### PR TITLE
everloop performance optimization

### PR DESCRIPTION
**Pull request by cmetz**
- switched from SpiWrite16 to SpiWriteBurst
- this reduces the cpu consumption e.g. in the everloop_demo from 13% to under 1%
- the casting from the image to an unsigned char * can possibly be made better (My knowledge of cpp dates back a long time)

### CODE REVIEW

File: cpp/driver/everloop.cpp [hunk 1]
```diff
@@ -29,22 +30,19 @@ Everloop::Everloop() {}
 bool Everloop::Write(const EverloopImage* led_image) {
   if (!wishbone_) return false;
 
-  uint16_t wb_data_buffer;
-  char* data_buffer = reinterpret_cast<char*>(&wb_data_buffer);
+  std::valarray<uint8_t> write_data (led_image->leds.size() * 4);
```
Comment (kdpatino) [inline cpp/driver/everloop.cpp]: You should use `std::valarray<unsigned char> write_data (led_image->leds.size() * 4);`

File: cpp/driver/everloop.cpp [hunk 2]
```diff
@@ -29,22 +30,19 @@ Everloop::Everloop() {}
...
-                          wb_data_buffer);
-
-    addr_offset = addr_offset + 2;
+    write_data[led_offset + 0] = led.red;
+    write_data[led_offset + 1] = led.green;
+    write_data[led_offset + 2] = led.blue;
+    write_data[led_offset + 3] = led.white;
+    led_offset += 4;
   }
+  wishbone_->SpiWriteBurst(kEverloopBaseAddress,
```
Comment (kdpatino) [inline cpp/driver/everloop.cpp]: And you don't need to cast the buffer: 
```
  wishbone_->SpiWriteBurst(kEverloopBaseAddress,
		           &write_data[0],
			   write_data.size());
```

Comment (cmetz) [reply cpp/driver/everloop.cpp]: thanks i have changed it.

Comment (kdpatino) [reply cpp/driver/everloop.cpp]: It's ready to merge :+1:

**Comment by kdpatino** [comment]
ThankYou @cmetz !!!

**Pull request merged**